### ISSUE TITLE
Restricted Jethro accounts should not be able to view 'no congregation' persons

### ISSUE THREAD

**Issue by jefft** [tbar0970/jethro-pmm#877]
Jethro allows accounts to be restricted in which persons they can view. For instance, this user should only be able to see members of the 'Musicians' group. They should not be able to see members of 9am and 5pm congregations:

![image](https://user-images.githubusercontent.com/205995/223903880-2682eb97-7b10-4e06-a447-293f4d049f1f.png)

However, what about persons not in any congregation, like visitors or members of youth groups?

I would have expected they would be invisible, but Jethro with the settings above shows them.

### Demo
Attached ([jethro_demosystem.sql.gz](https://github.com/tbar0970/jethro-pmm/files/10927340/jethro_demosystem.sql.gz)) is a sample Jethro database, containing Ben Churchgoer who attends 5pm, and his daughter Claire who is in no congregation but is member of 'Youth Group':

![image](https://user-images.githubusercontent.com/205995/223906891-fd1b4f4e-f233-4900-84f7-a79b43bdffab.png)

The `bandviewer` account is restricted to 'Musicians' as above, but can see not only the legitimate 'Musicians' members, but Claire too:
![image](https://user-images.githubusercontent.com/205995/223907884-0708a2e0-c22d-4c53-819b-675a5b27789e.png)

(password for `demo` and `bandviewer` accounts is `demo123123`)

**Comment by jefft** [comment]
Suggested fix:
- Add a 'no congregation' checkbox below defined congregations:
![image](https://user-images.githubusercontent.com/205995/223908243-114a2d59-7947-491b-a050-bde8baacc6a0.png)
- Change the default behaviour to exclude people not in explicitly chosen congregations.

**Comment by tbar0970** [comment]
I don't think we need to change the interface. Your assumption about the original case was correct. I'll investigate why the congregationless person was incorrectly showing up.

**Comment by jefft** [comment]
Well, say someone does actually want, say, "5pm plus congregationless visitors". Right now they just check "5pm".

If you fix the bug, and suddenly the congregationless people disappear, how do they get the original behaviour back? That's where we'd need a 'no congregation' checkbox.

**Comment by tbar0970** [comment]
Ability to see "5pm plus congregationless contacts" would be a new feature, which somebody could request if they needed it :)
Meanwhile there's an actual bug to fix first!

**Comment by tbar0970** [comment]
Oh right, I see what you mean - people might be relying on the current behaviour.
But I doubt it.  The explanatory text is pretty clear.

**Comment by tbar0970** [comment]
Diagnosis: Since https://github.com/tbar0970/jethro-pmm/commit/3e07f47b906c838b93b9d0abd7fa521873efcbc0#diff-0f35e6e15927e7239b1b988169fa734e54756c1174359534dbb3c188cc78a6b1R225 the view definition erroneously looked for a congregationid column in the account_group_restriction table.  Mysql treated this as a reference to person.congregationid, which was the wrong idea altogether.
